### PR TITLE
Added recipe for ‘reproject’ package

### PR DESCRIPTION
**Pull request by astrofrog**
This has C/Cython extensions - is there anything in particular I need to do?

### CODE REVIEW

File: recipes/reproject/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - cython
+
+  run:
+    - python
+    - numpy
```
Comment (mwcraig) [inline recipes/reproject/meta.yaml]: make this `numpy x.x` to build pin the numpy version during the build.

File: recipes/reproject/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+build:
+  number: 0
+  script: python setup.py install --offline --no-git --single-version-externally-managed --record record.txt
+
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - cython
```
Comment (mwcraig) [inline recipes/reproject/meta.yaml]: Adding `numpy` as a build requirement should fix the CI failures. My understanding is that it does *not* need to be `numpy x.x` here, just `numpy`.

**Comment by astrofrog** [comment]
This is ready for final review/merging

File: recipes/reproject/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    - numpy
+
+  run:
+    - python
+    - numpy x.x
```
Comment (ocefpaf) [inline recipes/reproject/meta.yaml]: My "rule of thumb" to check if you need `numpy x.x` is to check if numpy include is used in setup.py (not bullet proof but works most of the time). It looks like you don't need it here.

PS: Also, if you really needed it you must add it in the `run` and `build` sections.

File: recipes/reproject/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    - astropy >=1.0
```
Comment (ocefpaf) [inline recipes/reproject/meta.yaml]: We need an astropy recipe in conda-forge :smile:

**Comment by ocefpaf** [comment]
Some minor comments and this is good to go.  I noted that the failure before was:

```shell
Traceback (most recent call last):
  File "setup.py", line 119, in <module>
    **package_info
  File "/opt/conda/envs/_build/lib/python2.7/distutils/core.py", line 151, in setup
    dist.run_commands()
  File "/opt/conda/envs/_build/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/opt/conda/envs/_build/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/opt/conda/envs/_build/lib/python2.7/site-packages/setuptools-23.0.0-py2.7.egg/setuptools/command/install.py", line 61, in run
  File "/opt/conda/envs/_build/lib/python2.7/distutils/command/install.py", line 563, in run
    self.run_command('build')
  File "/opt/conda/envs/_build/lib/python2.7/distutils/cmd.py", line 326, in run_command
    self.distribution.run_command(command)
  File "/opt/conda/envs/_build/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/opt/conda/envs/_build/lib/python2.7/distutils/command/build.py", line 127, in run
    self.run_command(cmd_name)
  File "/opt/conda/envs/_build/lib/python2.7/distutils/cmd.py", line 326, in run_command
    self.distribution.run_command(command)
  File "/opt/conda/envs/_build/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/opt/conda/conda-bld/work/reproject-0.3/astropy_helpers/astropy_helpers/commands/build_ext.py", line 288, in run
    np_include = get_numpy_include_path()
  File "/opt/conda/conda-bld/work/reproject-0.3/astropy_helpers/astropy_helpers/utils.py", line 94, in get_numpy_include_path
    import numpy
ImportError: No module named numpy
Command failed: /bin/bash -x -e /opt/conda/conda-bld/work/reproject-0.3/conda_build.sh
```

That means `astropy_helpers` **does** need `numpy x.x` (see the `get_numpy_include_path`in that submodule).

I am not sure if you want to split `astropy_helpers` into another package or if you want to use `numpy x.x` here (in `reproject` recipe). The former is more "elegant" the latter, but is a low hanging fruit that would allow us to merge this recipe soon.

File: recipes/reproject/meta.yaml (hunk 3 above)
Comment (astrofrog) [inline recipes/reproject/meta.yaml]: I think we do - this package includes Cython extensions which including linking against Numpy?

**Comment by astrofrog** [comment]
@ocefpaf - astropy-helpers itself doesn't require numpy, but the requirement comes from the fact that astropy-helpers defines a build_ext command which we are using here to build the Cython extensions, and this in turn requires Numpy. So I think it doesn't make sense to add a separate package for astropy-helpers, but instead to make sure numpy is a build time requirement for reproject.

File: recipes/reproject/meta.yaml (hunk 3 above)
Comment (ocefpaf) [inline recipes/reproject/meta.yaml]: See my comment below. (I missed the `astropy_helpers` `submodule` in my first pass :smile:)

**Comment by ocefpaf** [comment]
>  but instead to make sure numpy is a build time requirement for reproject.

:+1:

You know your package better than me :smile: 

Just add `numpy x.x` in the run and build section and this recipe is good to go.

**Comment by astrofrog** [comment]
@ocefpaf - should be done!

**Comment by ocefpaf** [comment]
Thanks @astrofrog!

**Comment by mwcraig** [comment]
@ocefpaf -- isn't `numpy x.x` only required in run now that https://github.com/conda/conda-build/pull/650 has been merged?

**Comment by jakirkham** [comment]
I don't think that is in a release we use across the board yet. 😞

**Comment by ocefpaf** [comment]
> @ocefpaf -- isn't `numpy x.x` only required in run now that https://github.com/conda/conda-build/pull/650 has been merged?

I lost track of the `conda-build` releases, but to be on the safe side I still use (and recommend) `numpy x.x` at both build and run time. Thanks for bringing it up though. I will investigate this further and update the example!

**Comment by mwcraig** [comment]
Looking forward to sprinting on this kind of stuff next week!

**Pull request merged**